packages/validator/templates/OAS2OIS.json

```diff
@@ -0,0 +1,288 @@
+{
+	"__actions": [
+		{
+			"__insert": {
+				"__target": "oisFormat",
+				"__value": "1.0.0"
+			}
+		},
+		{
+			"__insert": {
+				"__target": "apiSpecifications.servers",
+				"__value": []
+			}
+		},
+		{
+			"__insert": {
+				"__target": "endpoints",
+				"__value": []
+			}
+		},
+		{
+			"__insert": {
+				"__target": "apiSpecifications.security",
+				"__value": {}
+			}
+		},
+		{
+			"__insert": {
+				"__target": "apiSpecifications.components.securitySchemes",
+				"__value": {}
+			}
+		},
+		{
+			"__insert": {
+				"__target": "apiSpecifications.paths",
+				"__value": {}
+			}
+		}
+	],
+	"info": {
+		"title": {
+			"__actions": [
+				{
+					"__copy": {
+						"__target": "title"
+					}
+				}
+			]
+		},
+		"version": {
+			"__actions": [
+				{
+					"__copy": {
+						"__target": "version"
+					}
+				}
+			]
+		},
+		"__ignore": {}
+	},
+	"servers": {
+		"__arrayItem": {
+			"url": {
+				"__regexp": "^(https?|ftp)://[^\\s/$.?#].[^\\s]*$",
+				"__actions": [
+					{
+						"__copy": {
+							"__target": "apiSpecifications.servers[0].url"
+						}
+					}
+				]
+			}
+		}
+	},
+	"paths": {
+		"__keyRegexp": "^\\/[^\\s'\"\\\\]+$",
+		"__objectItem": {
+			"__keyRegexp": "^(get|post)$",
+			"__objectItem": {
+				"__actions": [
+					{
+						"__insert": {
+							"__target": "apiSpecifications.paths.{{1}}.{{2}}.parameters",
+							"__value": []
+						}
+					},
+					{
+						"__insert": {
+							"__target": "endpoints[].name",
+							"__value": "{{1}}"
+						}
+					},
+					{
+						"__insert": {
+							"__target": "endpoints[_].operation.method",
+							"__value": "{{2}}"
+						}
+					},
+					{
+						"__insert": {
+							"__target": "endpoints[_].operation.path",
+							"__value": "{{1}}"
+						}
+					},
+					{
+						"__insert": {
+							"__target": "endpoints[_].reservedParameters",
```

Comment:
I think this should be initialized to an empty array. AFAIK, reserved parameters *do not* map to operation parameters (i.e. the parameters of the actual API call(s)). So, if we're converting from OAS to OIS, I think it would make sense to initialize the reserved params to an empty array, because by definition, the OAS wouldn't have any reserved params to specify.

https://github.com/api3dao/api3-docs/blob/8e41ab2597cb19cb75e4904d6e9b214287c1d738/airnode/ois.md#54-reservedparameters

Reply:
Or, you can make `reservedParameters` an optional key.

Reply:
I think I told Jakub to do this due to practical reasons. Adding all three reserved parameters means that the requester will be able to specify these, so it's similarly non-opinionated as not adding them at all. However, the requester will want to use these parameters in practice, so adding them here makes more sense.

We require the provider to explicitly add the reserved parameters for them to be enabled because we will have parameters in the future that the provider may want to disable (for example specifying off-chain computation).

Btw `reservedParameters` is a required field so would have to exist even if as an empty array.